tests/Fixer/FunctionNotation/PhpdocToParamTypeFixerTest.php

```diff
@@ -0,0 +1,288 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\FunctionNotation;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author Jan Gantzert <jan@familie-gantzert.de>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\FunctionNotation\PhpdocToParamTypeFixer
+ */
+final class PhpdocToParamTypeFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     * @param null|int    $versionSpecificFix
+     * @param null|array  $config
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix($expected, $input = null, $versionSpecificFix = null, $config = null)
+    {
+        if (
+            (null !== $input && \PHP_VERSION_ID < 70000)
+            || (null !== $versionSpecificFix && \PHP_VERSION_ID < $versionSpecificFix)
+        ) {
+            $expected = $input;
+            $input = null;
+        }
+        if (null !== $config) {
+            $this->fixer->configure($config);
+        }
+
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases()
+    {
+        return [
+            'no phpdoc param' => [
+                '<?php function my_foo() {}',
```

Comment:
For this and the next 3 cases: yes they are all invalid, however, there are no parameters to typehint anyway, so the fixer can't do anything anyway. Shouldn't they have a parameter?